### PR TITLE
[CORE24-114] task(frontend): frontend skeleton

### PR DESCRIPTION
**Pull request by internetti**
Jira ticket: JIRA-CORE24-114

## What have you implemented?
created a frontend skeleton using the nx generator with vite, cypress, sass and react

## How to test it?
nx run core-frontend:build
nx run core-frontend:preview

## Extra/next steps
tests are still having an issue

### CODE REVIEW

File: apps/core-frontend/project.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      "defaultConfiguration": "development",
+      "options": {
+          "buildTarget": "core-frontend:build"
+      },
+      "configurations": {
+        "development": {
+          "buildTarget": "core-api:build:development"
+        },
+        "production": {
+          "buildTarget": "core-api:build:production"
```
Comment (dfigueroa87) [inline apps/core-frontend/project.json:18]: Shouldn't these two be `core-frontend`?

File: apps/core-frontend/src/app/app.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+              <Link to="/">Click here to go back to root page.</Link>
+            </div>
+          }
+        />
+      </Routes>
+    </div>
+  );
+};
+
+export default App;
```
Comment (neb42) [inline apps/core-frontend/src/app/app.tsx:46]: On the backend setup we decided to use named exports only. So remove this line and add an export to line 3

File: apps/core-frontend/vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    },
+  },
+
+  optimizeDeps: {
+    esbuildOptions: {
+      resolveExtensions: extensions,
+      jsx: 'automatic',
+      loader: { '.js': 'jsx' },
+    },
+  },
```
Comment (neb42) [inline apps/core-frontend/vite.config.ts:55]: I don't think this is needed if we stop using react native

File: package.json [hunk 4]
```diff
@@ -2,45 +2,75 @@
...
     "url-loader": "^4.1.1",
+    "vite": "~5.0.0",
+    "vitest": "^1.3.1",
     "wait-on": "^7.2.0",
     "webpack-cli": "^5.1.4"
+  },
+  "type": "module",
+  "resolutions": {
+    "native-base": "3.4.18"
   }
```
Comment (neb42) [inline package.json:75]: Remove this

File: package.json [hunk 5]
```diff
@@ -2,45 +2,75 @@
...
     "ts-jest": "^29.1.0",
     "ts-node": "10.9.1",
     "typescript": "~5.3.2",
     "url-loader": "^4.1.1",
+    "vite": "~5.0.0",
+    "vitest": "^1.3.1",
     "wait-on": "^7.2.0",
     "webpack-cli": "^5.1.4"
+  },
+  "type": "module",
```
Comment (neb42) [inline package.json:72]: Remove this

File: package.json [hunk 6]
```diff
@@ -2,45 +2,75 @@
...
   "dependencies": {
+    "@expo-google-fonts/roboto": "~0.2.2",
     "axios": "^1.6.0",
     "express": "^4.18.1",
+    "jsx-runtime": "^1.2.0",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "react-native-safe-area-context": "3.3.2",
+    "react-native-svg": "12.1.1",
+    "react-native-web": "0.17.1",
```
Comment (neb42) [inline package.json:15]: Remove

File: package.json [hunk 7]
```diff
@@ -2,45 +2,75 @@
   "name": "@nrcno/core",
   "version": "0.0.0",
   "license": "MIT",
-  "scripts": {},
   "private": true,
   "dependencies": {
+    "@expo-google-fonts/roboto": "~0.2.2",
```
Comment (neb42) [inline package.json:7]: Remove

**Review by neb42 — CHANGES_REQUESTED** [review]
With the changes to `vite.config.ts` and the `module` field in `package.json`, you should be able to rename the `.cjs` files back to `.js`.

File: apps/core-frontend/project.json (hunk 1 above)
Comment (internetti) [reply apps/core-frontend/project.json:18]: of course. even after fixing this, it still seems to run the production as default, any ideas as to why?

Comment (dfigueroa87) [reply apps/core-frontend/project.json]: I'm guessing the `production` logged when running the build comes from vite and it's not related to this. Removing the production config and running the development build explicitly still works and logs the same

**Review by dfigueroa87 — COMMENTED** [review]
Looks good to me, you just seem to have some formatting issues in these files:
apps/core-frontend-e2e/tsconfig.json
apps/core-frontend/project.json

**Review by dfigueroa87 — APPROVED** [review]

**Review by neb42 — APPROVED** [review]

**Pull request merged**